rust/src/util/api.rs

```diff
@@ -32,16 +33,30 @@ pub fn init_log_with_file(file: File) -> Option<()> {
     }
 }
 
+// Serialize the GPU device names into a vector
```

Comment:
This could be a doc comment (even if it's a private function) with three leading slashes.

Reply:
Originally I had this as a doc comment, but changed it since it was only called internally.  If there preference it to have it as a doc comment I can switch it back.

Reply:
My take on it is: if it really documents the function, why not making it a doc comment even if it's internal.